python/data_structures/binary_tree.py

```diff
@@ -54,3 +54,33 @@ def post_order_recursive(node, current_list):
         post_order_list = []
         post_order_recursive(self.root, post_order_list)
         return post_order_list
+
+    def find_maximum_value(self):
+        def traverse_for_max(root, max=0):
+            if root is None:
+                return max
+            if max < root.value:
+                max = root.value
+
+            left_side_value = traverse_for_max(root.left, max)
+            if max < left_side_value:
+                max = left_side_value
+
+            right_side_value = traverse_for_max(root.right, max)
+            if max < right_side_value:
+                max = right_side_value
```

Comment:
Lines 65 through 71 could probably be simplified to `max = traverse_for_max(root.left, max)` and the same for right.

This is because your recursive function already performs the comparison for you and returns the max value between the node and whatever is passed in, which is the current max value.